src/main/java/com/google/cloud/flink/bigquery/common/DataWriterContext.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.flink.bigquery.common;
+
+import java.io.IOException;
+
+/** An internal version to Flink DataSource DataWriter interface */
+public interface DataWriterContext<T> {
+  void write(T row) throws IOException;
+
+  WriterCommitMessageContext commit() throws IOException;
+
+  void commitFinalizedStream() throws IOException;
```

Comment:
what is the difference between this and commit() ? in the implementation `commitFinalizedStream()` does even less than commit

Reply:
commit() method in writer helper class is only for  finalizing the stream, it is not commiting the data to the table. 
This writer helper commit() method is calling through this commit method.

commitFinalizedStream() is implemented for committing the finalized stream to the table. This part not found in the writer helper class. So added this method